optaplanner-examples/src/test/java/org/optaplanner/examples/common/persistence/OpenDataFilesTest.java

```diff
@@ -30,23 +31,24 @@
     protected abstract CommonApp<Solution_> createCommonApp();
 
     private static List<File> getSolutionFiles(CommonApp<?> commonApp) {
-        List<File> fileList = new ArrayList<>(0);
-        File dataDir = CommonApp.determineDataDir(commonApp.getDataDirName());
-        File unsolvedDataDir = new File(dataDir, "unsolved");
-        if (!unsolvedDataDir.exists()) {
-            throw new IllegalStateException("The directory unsolvedDataDir (" + unsolvedDataDir.getAbsolutePath()
-                    + ") does not exist.");
+        try (SolutionBusiness<?, ?> solutionBusiness = commonApp.createSolutionBusiness()) {
+            File unsolvedDataDir = solutionBusiness.getUnsolvedDataDir();
+            if (!unsolvedDataDir.exists()) {
+                throw new IllegalStateException("The directory unsolvedDataDir (" + unsolvedDataDir.getAbsolutePath()
+                        + ") does not exist.");
+            }
+            String inputFileExtension = commonApp.createSolutionFileIO().getInputFileExtension();
+            List<File> fileList = new ArrayList<>(
+                    getAllFilesRecursivelyAndSorted(unsolvedDataDir, file -> file.getName().endsWith(inputFileExtension)));
+            File solvedDataDir = solutionBusiness.getSolvedDataDir();
+            if (solvedDataDir.exists()) {
+                String outputFileExtension = commonApp.createSolutionFileIO().getOutputFileExtension();
```

Comment:
We already call `commonApp.createSolutionFileIO()` a couple of lines above. Let's keep a reference to the `SolutionFileIO`.